ipfx/dataset/ephys_nwb_data.py

```diff
@@ -192,37 +176,30 @@ def get_session_start_time(self):
 
         return datetime_object
 
-    def get_recording_date(self):
-
-        datetime_object = self.get_session_start_time()
-
-        recording_date = datetime_object.strftime("%Y-%m-%d %H:%M:%S")
 
-        return recording_date
+    def get_sweep_record(self, sweep_number: int):
+        raise NotImplementedError
 
-    def get_stimulus_unit(self, sweep_number):
+    def get_stimulus_unit(self,sweep_number):
 
-        sweep_map = self.get_sweep_map(sweep_number)
+        series = self.nwb.sweep_table.get_series(sweep_number)
 
-        with h5py.File(self.nwb_file, 'r') as f:
-            sweep_stimulus = f[self.stimulus_path][sweep_map["stimulus_group"]]
-            stimulus_dataset = sweep_stimulus["data"]
+        if series is None:
+            raise ValueError("No TimeSeries found for sweep number {}.".format(sweep_number))
 
-            unit = self.get_unit_name(stimulus_dataset.attrs)
-            unit_str = self.get_long_unit_name(unit)
+        stimulus = []
 
-            return unit_str
+        for s in series:
+            if isinstance(s,(CurrentClampStimulusSeries,VoltageClampStimulusSeries)):
```

Comment:
is there a common base type?

Reply:
There is (PatchClampSeries), but it is also the parent for the CurrentClampSeries, VoltageClampSeries used by the response series.